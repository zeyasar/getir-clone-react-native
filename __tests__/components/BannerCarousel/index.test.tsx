import React from "react";
import { render } from "@testing-library/react-native"
import BannerCarousel from "../../../src/components/BannerCarousel";


describe("components/BannerCarousel", () => {
    test('renders correctly', () => {
        const {toJSON} = render(<BannerCarousel/>)
        expect(toJSON).toMatchSnapshot()
    })

    test("renders children number correctly", () => {
        const {getByTestId} = render(<BannerCarousel />)
        const banner = getByTestId('banner-carousel')
        expect(banner.props.data.length).toBe(4)
    
    })

    test("renders props types correctly ",() => {
        const {getByTestId} = render(<BannerCarousel />)
        const banner = getByTestId('banner-carousel')

        expect(banner.props.decelerationRate).toBe("fast")
    })
    test("render cdn images in banner carousel",() => {
        const {getByTestId} = render(<BannerCarousel />)
        const banner = getByTestId('banner-carousel')
        const bannerCarouselItems = banner.props.data

        bannerCarouselItems.map((item,index) => {
            const cdnElement:string = item
            expect(cdnElement.substring(0,11)).toEqual("https://cdn")
        })
        
    })
})

