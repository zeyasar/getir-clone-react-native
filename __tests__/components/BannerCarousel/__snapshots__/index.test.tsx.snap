// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BannerCarousel renders correctly 1`] = `[Function]`;
